modules/cmsis_dsp/CMakeLists.txt

```diff
@@ -0,0 +1,931 @@
+# Copyright (c) 2023 Meta Platforms
+# SPDX-License-Identifier: Apache-2.0
+
+if(CONFIG_CMSIS_DSP)
+
+  set(CMSIS_DSP_DIR ${ZEPHYR_CURRENT_MODULE_DIR})
+  set(cmsis_glue_path ${ZEPHYR_CMSIS_MODULE_DIR})
+
+  zephyr_library()
+
+  zephyr_library_compile_options(-Ofast)
+
+  zephyr_include_directories(
+    ${CMSIS_DSP_DIR}/Include
+  )
+
+  zephyr_library_include_directories(
+    ${CMSIS_DSP_DIR}/PrivateInclude
+    ${cmsis_glue_path}/CMSIS/Core/Include
+  )
+
+  # Global Feature Definitions
+  zephyr_library_compile_definitions_ifdef(CONFIG_CMSIS_DSP_NEON                  ARM_MATH_NEON)
+  zephyr_library_compile_definitions_ifdef(CONFIG_CMSIS_DSP_NEON_EXPERIMENTAL     ARM_MATH_NEON_EXPERIMENTAL)
+  zephyr_library_compile_definitions_ifdef(CONFIG_CMSIS_DSP_LOOPUNROLL            ARM_MATH_LOOPUNROLL)
+  zephyr_library_compile_definitions_ifdef(CONFIG_CMSIS_DSP_ROUNDING              ARM_MATH_ROUNDING)
+  zephyr_library_compile_definitions_ifdef(CONFIG_CMSIS_DSP_MATRIXCHECK           ARM_MATH_MATRIX_CHECK)
+  zephyr_library_compile_definitions_ifndef(CONFIG_CMSIS_DSP_FLOAT16              DISABLEFLOAT16)
+  zephyr_library_compile_definitions_ifdef(CONFIG_CMSIS_DSP_AUTOVECTORIZE         ARM_MATH_AUTOVECTORIZE)
+
+  # Helium is automatically enabled by CMSIS-DSP if the processor supports it
+  zephyr_library_compile_definitions_ifdef(CONFIG_CMSIS_DSP_HELIUM_EXPERIMENTAL   ARM_MATH_HELIUM_EXPERIMENTAL)
+
+  zephyr_compile_definitions_ifndef(CONFIG_ARM __GNUC_PYTHON__)
+
+  if (CONFIG_CMSIS_DSP_NEON OR CONFIG_CMSIS_DSP_NEON_EXPERIMENTAL)
+    zephyr_library_include_directories("${CMSIS_DSP_DIR}/ComputeLibrary/Include")
+  endif()
+
+  if (CONFIG_ARMV8_1_M_MVEI OR CONFIG_ARMV8_1_M_MVEF)
+    if (CONFIG_CMSIS_DSP_LAX_VECTOR_CONVERSIONS)
+      zephyr_library_compile_options($<$<STREQUAL:${CMAKE_C_COMPILER_ID},GNU>:-flax-vector-conversions>)
+      zephyr_library_compile_options($<$<STREQUAL:${CMAKE_C_COMPILER_ID},ARMClang>:-flax-vector-conversions=integer>)
+    else()
+      zephyr_library_compile_options($<$<STREQUAL:${CMAKE_C_COMPILER_ID},GNU>:-fno-lax-vector-conversions>)
+      zephyr_library_compile_options($<$<STREQUAL:${CMAKE_C_COMPILER_ID},ARMClang>:-flax-vector-conversions=none>)
+    endif()
+  endif()
+
+  zephyr_library_compile_options(-Ofast)
+
+  zephyr_compile_definitions_ifndef(CONFIG_ARM __GNUC_PYTHON__)
+
+  # BasicMathFunctions
+
+  set(SRCF64 ${CMSIS_DSP_DIR}/Source/BasicMathFunctions/arm_abs_f64.c
```

Comment:
Is there a reason you are manually listing all the files instead of just globbing everything in `${CMSIS_DSP_DIR}/Source/BasicMathFunctions`?

Reply:
I could :shrug: ... but that is just how the upstream does it